docs/model-dev-guide/profiling.rst

```diff
@@ -0,0 +1,110 @@
+.. _profiling:
+
+###########
+ Profiling
+###########
+
+Optimizing a model's performance is often a trade-off between accuracy, time, and resource
+requirements. Training deep learning models is a time and resource intensive process, where each
+iteration can take several hours and accumulate heavy hardware costs. Though sometimes this cost is
+inherent to the task, unnecessary resource consumption can be caused by suboptimal code or bugs.
+Thus, achieving optimal model performance requires an understanding of how your model interacts with
+the system's computational resources.
+
+Profiling collects metrics on how computational resources like CPU, GPU, and memory are being
+utilized during a training job. It can reveal patterns in resource utilization that indicate
+performance bottlenecks and pinpoint areas of the code or pipeline that are causing slowdowns or
+inefficiencies.
+
+Profiling a training job can be instrumented at many different layers, from generic system-level
+metrics to individual model operators and GPU kernels. Determined provides a few options for
+profiling, each targeting a different layer in a training job at various levels of detail:
+
+-  :ref:`Determined system metrics profiler <how-to-profiling-det-profiler>` collects general
+   system-level metrics and provides an overview of hardware usage during an experiment.
+-  :ref:`Native profiler integration <how-to-profiling-native-profilers>` enables model profiling in
+   training APIs that provides fine-grained metrics specific to your model.
+-  :ref:`Prometheus/Grafana integration <how-to-profiling-prom-grafana>` can be set up to track
+   detailed hardware metrics and monitor overall cluster health.
+
+.. _how-to-profiling:
+
+.. _how-to-profiling-det-profiler:
+
+*********************
+ Determined Profiler
```

Comment:
Is it appropriate to provide references here for how profiler metrics can be inspected?

Reply:
Mmm. I missed this part.

> The Determined profiler collects a set of system metrics throughout an experiment which can be visualized in the WebUI under the experiment’s “Profiler” tab. It is supported for all training APIs, but is not enabled by default.

My fault. Like usual, I'd like someone else to blame when a thing was my fault. In that respect, could / should it have been clearer?